File: Akureyri/akureyri.c
#include <stdio.h>
#include <string.h>

int main() {
    int N;
    scanf("%d", &N);

    char locations[10000][101];
    int counts[10000], distinct = 0;

    for (int i = 0; i < N; i++) {
        char name[101], location[101];
        scanf("%s", name);
        scanf("%s", location);

        int found = -1;

        for (int j = 0; j < distinct; j++) {
            if (strcmp(locations[j], location) == 0) {
                found = j;
                break;
            }
        }

        if (found == -1) {
            strcpy(locations[distinct], location);
            counts[distinct] = 1;
            distinct++;
        } else {
            counts[found]++;
        }
    }

    for (int i = 0; i < distinct; i++) {
        printf("%s %d\n", locations[i], counts[i]);
    }

    return 0;
}

File: Akureyri/akureyri.java
import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class akureyri {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();

        Map<String, Integer> counts = new HashMap<>();

        for (int i = 0; i < N; i++) {
            String name = scanner.next();
            String location = scanner.next();

            counts.put(location, counts.getOrDefault(location, 0) + 1);
        }

        for (Map.Entry<String, Integer> entry : counts.entrySet()) {
            System.out.println(entry.getKey() + " " + entry.getValue());
        }
    }
}
